misc/install_initrd.sh

```diff
@@ -0,0 +1,221 @@
+		shift # past argument=value
+	;;
+	--uninstall)
+		uninstall=1
+		break
+		# shift # past argument with no value
+	;;
+	--boot=*)
+		boot_device="${i#*=}"
+		shift # past argument=value
+	;;
+	--size=*)
+		ramdisk_size="${i#*=}"
+		shift # past argument=value
+	;;
+	--force)
+		force=1
+		shift # past argument with no value
+	;;
+	*)
+	        myerror "Error: Unknown argument."
+		# unknown option
+	;;
+esac
+done
+
+if ! whoami | grep root 2>/dev/null 1>/dev/null ; then
+	myerror "Sorry, this must be run as root."
+fi
+
+cwd="$(pwd)"
+
+if [ -z $kernel_version ] ; then
+	myerror "Error: missing argument '--kernel'."
+fi
+
+if [ -z $uninstall ] ; then
+
+	if [ -z $ramdisk_size ] ; then
+		myerror "Error: missing argument '--size'."
+	fi
+
+	if [ -z $boot_device ] ; then
+		myerror "Error: missing argument '--boot'."
+	fi
+	
+	if ! is_num "$ramdisk_size" ; then
+		myerror "Error: The ramdisk size must be a positive integer. Exiting..."
+	fi
+fi
+
+
+# Check for CentOS
+if hostnamectl | grep "CentOS Linux" >/dev/null 2>/dev/null ; then
+	echo " - CentOS detected!"
+
+	if [ -z $uninstall ] ; then
+
+		echo " - Installing by copying 96rapiddisk into /usr/lib/dracut/modules.d/..."
+		rm -rf /usr/lib/dracut/modules.d/96rapiddisk
+		cp -rf 96rapiddisk /usr/lib/dracut/modules.d/
+		echo " - Editing /usr/lib/dracut/modules.d/96rapiddisk/run_rapiddisk.sh..."
+		sed -i 's/RAMDISKSIZE/'${ramdisk_size}'/' /usr/lib/dracut/modules.d/96rapiddisk/run_rapiddisk.sh
+		sed -i 's,BOOTDEVICE,'${boot_device}',' /usr/lib/dracut/modules.d/96rapiddisk/run_rapiddisk.sh
+		echo " - chmod +x /usr/lib/dracut/modules.d/96rapiddisk/* ..."
+		chmod +x /usr/lib/dracut/modules.d/96rapiddisk/*
+	else
+		echo " - Uninstalling by removing dir /usr/lib/dracut/modules.d/96rapiddisk..."
+		rm -rf /usr/lib/dracut/modules.d/96rapiddisk
+	fi
+	
+	echo " - Running 'dracut --kver $kernel_version -f'"
+	#dracut --add-drivers "rapiddisk rapiddisk-cache" --kver "$kernel_version" -f
+	dracut --kver "$kernel_version" -f
+	
+	echo " - Done under CentOS."
+	exit 0
+fi
+
+# Ubuntu
+
+if [ -d /etc/initramfs-tools/hooks ] && [ -d /etc/initramfs-tools/scripts/init-premount ] ; then
+	hook_dest=/etc/initramfs-tools/hooks/rapiddisk_hook
+	bootscript_dest=/etc/initramfs-tools/scripts/init-premount/rapiddisk
+elif [ -d /usr/share/initramfs-tools/hooks ] && [ -d /usr/share/initramfs-tools/scripts/init-premount ] ; then
+	hook_dest=/usr/share/initramfs-tools/hooks/rapiddisk_hook
+	bootscript_dest=/usr/share/initramfs-tools/scripts/init-premount/rapiddisk
+else
+	myerror "Error: I can't find any suitable place for initramfs scripts."
+fi
+
+
+if [ ! -z $uninstall ] ; then
+	echo " - Starting uninstall process..."
+	if [ -x "$hook_dest" ] || [ -x "$bootscript_dest" ] ; then
+		echo " - Uninstalling scripts..."
+		rm -f "$hook_dest" 2>/dev/null
+		rm -f "$bootscript_dest" 2>/dev/null
+	fi
+	#if lsinitramfs /boot/initrd.img-"$kernel_version" | grep rapiddisk >/dev/null 2>/dev/null ; then
```

Comment:
Why is this commented out? Do we still need it here?

Reply:
Is commented out because I don't know if it really useful... yet!